commands/operator-sdk/cmd/generate/k8s.go

```diff
@@ -52,69 +53,130 @@ func k8sFunc(cmd *cobra.Command, args []string) {
-	if err != nil {
-		log.Fatalf("failed to perform code-generation: (%v)", err)
+	for _, gd := range genDirs {
+		err := runGoBuildCodegen(binDir, codegenSrcDir, gd)
+		if err != nil {
+			log.Fatal(err)
+		}
 	}
+}
 
-	log.Info("Code-generation complete.")
+func runGoBuildCodegen(binDir, repoDir, genDir string) error {
+	binPath := filepath.Join(binDir, filepath.Base(genDir))
+	installCmd := exec.Command("go", "build", "-o", binPath, genDir)
+	installCmd.Dir = repoDir
+	isVerbose := false
+	if gf, ok := os.LookupEnv("GOFLAGS"); ok && len(gf) != 0 {
+		installCmd.Env = append(os.Environ(), "GOFLAGS="+gf)
+		if strings.Contains(gf, "-v") {
+			isVerbose = true
+		}
+	}
+	if isVerbose {
+		installCmd.Stdout = os.Stdout
+		installCmd.Stderr = os.Stderr
+	} else {
+		installCmd.Stdout = ioutil.Discard
+		installCmd.Stderr = ioutil.Discard
+	}
+	return installCmd.Run()
 }
 
-// getGroupVersions parses the layout of pkg/apis to return the API groups and versions
-// in the format "groupA:v1,v2 groupB:v1 groupC:v2",
-// as required by the generate-groups.sh script
-func parseGroupVersions() (string, error) {
-	var groupVersions string
+// parseGroupVersions parses the layout of pkg/apis to return a map of
+// API groups to versions.
+func parseGroupVersions() (map[string][]string, error) {
+	gvs := make(map[string][]string)
 	groups, err := ioutil.ReadDir(scaffold.ApisDir)
 	if err != nil {
-		return "", fmt.Errorf("could not read pkg/apis directory to find api Versions: %v", err)
+		return nil, fmt.Errorf("could not read pkg/apis directory to find api Versions: %v", err)
 	}
 
 	for _, g := range groups {
 		if g.IsDir() {
 			groupDir := filepath.Join(scaffold.ApisDir, g.Name())
 			versions, err := ioutil.ReadDir(groupDir)
 			if err != nil {
-				return "", fmt.Errorf("could not read %s directory to find api Versions: %v", groupDir, err)
+				return nil, fmt.Errorf("could not read %s directory to find api Versions: %v", groupDir, err)
 			}
 
-			groupVersion := ""
+			gvs[g.Name()] = make([]string, 0)
 			for _, v := range versions {
 				if v.IsDir() && scaffold.ResourceVersionRegexp.MatchString(v.Name()) {
-					groupVersion = groupVersion + v.Name() + ","
+					gvs[g.Name()] = append(gvs[g.Name()], v.Name())
 				}
 			}
-			groupVersions += fmt.Sprintf("%s:%s ", g.Name(), groupVersion)
 		}
 	}
 
-	if groupVersions == "" {
-		return "", fmt.Errorf("no groups or versions found in %s", scaffold.ApisDir)
+	if len(gvs) == 0 {
+		return nil, fmt.Errorf("no groups or versions found in %s", scaffold.ApisDir)
 	}
+	return gvs, nil
+}
 
-	return groupVersions, nil
+// createFQApis return a string of all fully qualified pkg + groups + versions
+// of pkg and gvs in the format:
+// "pkg/groupA/v1,pkg/groupA/v2,pkg/groupB:v1"
+func createFQApis(pkg string, gvs map[string][]string) (fqStr string) {
+	gn := 0
+	for g, vs := range gvs {
+		for vn, v := range vs {
+			fqStr += filepath.Join(pkg, g, v)
+			if vn < len(vs)-1 {
+				fqStr += ","
```

Comment:
nit: should we use `strings.Builder` instead of `+=` to be a bit more efficient with memory allocations?